src/wallet/test/rpc_wallet_tests.cpp

```diff
@@ -315,8 +315,25 @@ BOOST_AUTO_TEST_CASE(rpc_wallet)
         }
     };
 
+    auto check_lockbox_streams = [](UniValue obj, std::vector<std::string> recipients, std::vector<double> amounts) {
+        size_t n = recipients.size();
+        BOOST_REQUIRE_EQUAL(amounts.size(), n);
+        UniValue lockboxstreams = find_value(obj, "lockboxstreams");
+        BOOST_CHECK_EQUAL(lockboxstreams.size(), n);
+        if (lockboxstreams.size() != n) return;
```

Comment:
Philosophical conundrum: does the difference between `BOOST_REQUIRE` and `BOOST_CHECK` reflect semantics ("the test requires this" vs "the test is intended to check this"), or just the practical issue that the test shouldn't proceed past a failed `BOOST_REQUIRE`? If it's the former then this is correct. If the latter then it's overcomplicated and could just be `BOOST_REQUIRE_EQUAL(lockboxstreams.size(), n);`. But I'm fine with it as-is.